pkg/mimir/modules.go

```diff
@@ -702,6 +704,26 @@ func (t *Mimir) initQueryFrontendCodec() (services.Service, error) {
 	return nil, nil
 }
 
+// initQueryFrontendTopicOffsetsReader instantiates the topic offsets reader used by the query-frontend
+// when the ingest storage is enabled.
+func (t *Mimir) initQueryFrontendTopicOffsetsReader() (services.Service, error) {
+	if !t.Cfg.IngestStorage.Enabled {
```

Comment:
this means that we should also change the migration procedure, right? At the very least query-frontends should get ingest storage enabled on them along with queriers. And maybe ideally enabled before queriers, so that we don't fall back to relying on ingesters fetching the offsets

Reply:
> this means that we should also change the migration procedure, right? At the very least query-frontends should get ingest storage enabled on them along with queriers.

Yes, absolutely. Will do in a separate PR.

> And maybe ideally enabled before queriers, so that we don't fall back to relying on ingesters fetching the offsets

This is not required. It doesn't matter the rollout order. If offsets are not propagated by query-frontend to ingesters, it will keep working as of today.